ci: enable freeze-on-ready and update sdk-release skill (#4313)
OpenHands/software-agent-sdk: .agents/skills/sdk-release/SKILL.md

This skill walks through the software-agent-sdk release process step by step.

The SDK releases via the shared **[OpenHands/release-actions](https://github.com/OpenHands/release-actions)**

> (release-please) automation. You no longer pick a version or create a release

Reviewer: ```suggestion
(release-please) automation. You must not pick a version or create a release
```
